lib/hexpm_web/controllers/oauth_login_controller.ex

```diff
@@ -0,0 +1,81 @@
+defmodule HexpmWeb.OAuthLoginController do
+  use HexpmWeb, :controller
+
+  require Logger
+
+  plug :nillify_params, ["return"]
+
+  def create(conn, %{"code" => code}) do
+    with {:ok, access_token} <- Hexpm.OAuthProviders.GitHub.get_access_token(code),
+         {:ok, %{id: user_id, email: email}} <-
+           Hexpm.OAuthProviders.GitHub.get_user(access_token),
+         user <- Hexpm.Accounts.Users.get_by_github_id(user_id) do
+      forward_to_login_or_sign_up(conn, user, user_id, email)
+    else
+      error ->
+        Logger.error("Error when trying to create OAuthLogin: #{inspect(error)}")
+
+        conn
+        |> put_flash(:error, "Something wrong happened, try again later")
```

Comment:
```suggestion
        |> put_flash(:error, "Something went wrong, try again later")
```
What is most likely to have gone wrong in this? Can we give the user a better idea of what to correct?

Have you seen this style in the `with` where you wrap each line in a tuple to better know how it failed (and perhaps give a better error message?
```
{:get_access_token, {:ok, access_token}} <- {:get_access_token, Hexpm.OAuthProviders.GitHub.get_access_token(code)},
```
and in the else block, match it:
```
{:get_access_token, _} -> "GitHub could not validate your access token, please try again"
```
WDYT?

Reply:
Basically, flawed connection with GitHub.

I think your code is better and I'm gonna change to something like it